data-processing/entities/citations/commands/resolve_bibitems.py

```diff
@@ -118,3 +130,45 @@ def save(self, item: MatchTask, result: BibitemMatch) -> None:
 
         resolutions_path = os.path.join(resolutions_dir, "resolutions.csv")
         file_utils.append_to_csv(resolutions_path, result)
+
+    @staticmethod
+    def similarity_count_vectorizer(s2_return: str, bibliography_reference: str) -> float:
+        """
+        Similarity measure based on CountVectorizer
+        https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text
+        .CountVectorizer.html#sklearn.feature_extraction.text.CountVectorizer.fit_transform
+        Implementation is using n_gram in range of 1-7. vectorizer is trained on the first parameter
+        which is the concatenated API return.
+        transform call returns a binary vector, each value corresponds to the dictionary entries in the
+        extracted_text_key
+        @param s2_return: Concatenated s2 return
+        @param bibliography_reference: Bibliography reference trying to match
+        @return:
+        """
+        if not s2_return or not bibliography_reference:
+            return 0.0
+
+        vectorizer = CountVectorizer(binary=True, ngram_range=(1, 5), stop_words='english')
+        try:
+            vectorizer.fit([s2_return])
+        except ValueError as e:
+            print(e)
+            return 0.0
+
+        if len(vectorizer.get_feature_names()) > 3:
```

Comment:
Yes that is right I have a section in the [doc](https://docs.google.com/document/d/1qbu3_3WOTCyPqeFPh_znHY8lVhxE5zygJNQeNjExLbU/edit#bookmark=id.xs6bjya1vntv) on that. If the vocabulary size is small < 3 similarity score is quite high and has only 2 value 1 and ~0.4. We want to avoid such S2 entries as probably we will make mistakes matching to them.